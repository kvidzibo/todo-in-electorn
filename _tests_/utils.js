import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

export const render = (comp, mode = 3) => {
  switch (mode) {
    case 1:
      return shallow(comp)
    case 2:
      return renderer.create(comp).toJSON()
    case 3:
      return {
        component: shallow(comp),
        tree: renderer.create(comp).toJSON()
      }
    case 4:
      return mount(comp)
    case 5:
      return {
        component: mount(comp),
        tree: renderer.create(comp).toJSON()
      }
  }
}
