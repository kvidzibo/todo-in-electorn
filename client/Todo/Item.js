import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Field, Button } from '../Components'
import Task from './Module'

class Item extends PureComponent {
  done = () => {
    const { item } = this.props
    Task.save({...item, status: 'done'})
  }

  remove = () => {
    const { item: { _id } } = this.props
    Task.remove({ _id })
  }

  render () {
    const { item } = this.props
    return (
      <div key={item._id} className="box">
        <Field>
          {item.task}
        </Field>
        <Field grouped uncontrolled>
          <Button control icon="check" onClick={this.done} />
          <Button control icon="times" onClick={this.remove} />
        </Field>
      </div>
    )
  }
}

Item.propTypes = {
  item: PropTypes.shape({
    _id: PropTypes.string,
    task: PropTypes.string
  })
}

export default Item
