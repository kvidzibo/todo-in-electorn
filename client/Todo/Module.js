import { remote } from 'electron'
const { Task } = remote.require('./app/modules')

export default Task
