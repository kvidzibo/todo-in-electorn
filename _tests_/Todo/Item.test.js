import React from 'react'
import { render } from '../utils'
import Item from '../../client/Todo/Item'
import Module from '../../client/Todo/Module'

jest.mock('electron', () => ({
  remote: {require: jest.fn(() => ({Task: {}}))}
}))
jest.mock('../../client/Todo/Module')

let item = {
  _id: 'testID',
  task: 'test task'
}

beforeEach(() => {
  Module.save = jest.fn().mockResolvedValue('')
  Module.remove = jest.fn().mockResolvedValue('')
})

test('renders correctly', () => {
  const { component, tree } = render(<Item item={item}/>, 5)
  expect(component.find('Field').length).toBe(2)
  expect(component.find('Field').first().text()).toBe('test task')
  expect(tree).toMatchSnapshot()
})

test('calls save correctly on done click', () => {
  const { component, tree } = render(<Item item={item}/>, 5)
  expect(component.find('Field').length).toBe(2)
  component.find('button').at(0).simulate('click')
  expect(Module.save.mock.calls.length).toBe(1)
  expect(Module.save.mock.calls[0][0]).toEqual({...item, status: 'done'})
  expect(tree).toMatchSnapshot()
})

test('calls remove correctly', () => {
  const { component, tree } = render(<Item item={item}/>, 5)
  expect(component.find('Field').length).toBe(2)
  component.find('button').at(1).simulate('click')
  expect(Module.remove.mock.calls.length).toBe(1)
  expect(Module.remove.mock.calls[0][0]._id).toBe(item._id)
  expect(tree).toMatchSnapshot()
})
