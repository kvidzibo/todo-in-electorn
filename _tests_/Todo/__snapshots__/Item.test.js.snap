// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls remove correctly 1`] = `
<div
  className="box"
>
  <div
    className="field"
  >
    <div
      className="control"
    >
      test task
    </div>
  </div>
  <div
    className="field is-grouped"
  >
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-check"
          />
        </span>
      </button>
    </div>
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-times"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`calls save correctly on done click 1`] = `
<div
  className="box"
>
  <div
    className="field"
  >
    <div
      className="control"
    >
      test task
    </div>
  </div>
  <div
    className="field is-grouped"
  >
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-check"
          />
        </span>
      </button>
    </div>
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-times"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="box"
>
  <div
    className="field"
  >
    <div
      className="control"
    >
      test task
    </div>
  </div>
  <div
    className="field is-grouped"
  >
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-check"
          />
        </span>
      </button>
    </div>
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fa fa-times"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
